web/src/lib/utils/slideshow-history.ts

```diff
@@ -0,0 +1,40 @@
+export class SlideshowHistory {
+  private slideshowHistory: string[] = [];
+  private slideshowHistoryIndex = 0;
+
+  constructor(private onNavigate: (assetId: string) => void) {}
+
+  reset(assetId: string) {
```

Comment:
This seems odd to me. Would it be better to use `reset()` followed by `append(assetId)` instead?